### PR TITLE
Cordova update due to Crosswalk interface change.

### PR DESCRIPTION
**Pull request by d0ngjun**
1. put all CordovaWebView creation code into
CordovaActivity.onXWalkReady method.

2. add CordovaApplication class.

### CODE REVIEW

**Comment by d0ngjun** [comment]
@halton @wang16 please help review.

Thanks.

**Comment by halton** [comment]
lgtm to me, I'll wait for @wang16 @lincsoon  review  before merging.

File: bin/templates/project/Activity.java [hunk 1]
```diff
@@ -25,10 +25,9 @@ Licensed to the Apache Software Foundation (ASF) under one
 public class __ACTIVITY__ extends CordovaActivity
 {
     @Override
-    public void onCreate(Bundle savedInstanceState)
```
Comment (wang16) [inline bin/templates/project/Activity.java]: Does here still need a `onCreate` override?

Comment (d0ngjun) [inline bin/templates/project/Activity.java]: CordovaActivity.onCreate has been called super.onCreate(savedInstanceState), seems we don't need it now.

File: framework/src/org/apache/cordova/CordovaActivity.java [hunk 2]
```diff
@@ -23,11 +23,10 @@ Licensed to the Apache Software Foundation (ASF) under one
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
-import org.apache.cordova.CordovaInterface;
-import org.apache.cordova.CordovaPlugin;
-import org.apache.cordova.LOG;
 import org.json.JSONException;
 import org.json.JSONObject;
+import org.xwalk.core.XWalkActivity;
+import org.xwalk.core.XWalkCoreWrapper;
```
Comment (lincsoon) [inline framework/src/org/apache/cordova/CordovaActivity.java]: It seems that XWalkCoreWrapper is not used. I also do not recommend to use it outside of XWalkAcvitiy.
XWalkCoreWrapper is not designed to be part of Embedding API, so will not be exposed to external in the future.

**Comment by lincsoon** [comment]
@d0ngjun In shared mode, the call process of first run will be onCreate -> onStart -> onResume -> pop up a dialog to prompt user to install xwalk library-> install finished ->onXWalkReady. So the appView is not ready yet when onStart and onResume is called. Is there any potential risk in this case?

**Comment by d0ngjun** [comment]
@lincsoon we do not assume appView is available in activity lifecycle callbacks, it's protected by carefully check:
        if (this.appView == null) {
            return;
        }

**Comment by wang16** [comment]
lgtm, thanks.

**Comment by lincsoon** [comment]
LGTM

**Pull request merged**